grow/pods/documents.py

```diff
@@ -264,6 +264,13 @@ def locales(self):
             if codes is None:
                 return []
             return locales.Locale.parse_codes(codes)
+        # If `$localization` is missing, use locales inferred from document
```

Comment:
I’d definitely expect it to _replace_ rather than append.

Common case we’d want to avoid:

* Main locales for project have been listed in podspec.yaml
* No locales defined in blueprint
* A document part declares a specific string for a single locale only
* `grow extract --localized` extracts this for _all_ podspec locales
* Thus this string appears as a missing translation in all locales, when in reality it only needs to be translated for one.

This is basically the bug reported in #157 — which exactly what this whole effort started off trying to solve.

Similarly, the list of main locales might be defined in the blueprint rather than podspec. So yep, I think really it should replace rather than append.

Which test(s) disagreed with this?